### PR TITLE
Add proj toolchain conf, localnet mkdir and peer auto discovery

### PR DESCRIPTION
**Pull request by Rubberbandits**
## Description

Ensure all compiles of subtensor are using a consistent toolchain, along with detecting if chainspec folde exists and creating it, and enabling auto-peer discovery for localnet nodes.

______

## Checklist

- [ ] ~Updated relevant documentation~
- [ ] ~Updated relevant tests~
- [ ] ~Tracked `rustfmt` changes~
- [X] No _new_ compiler warnings

### CODE REVIEW

File: scripts/localnet.sh [hunk 1]
```diff
@@ -44,8 +50,8 @@ bob_start=(
 	--ws-port 9947
 	--rpc-port 9935
 	--validator
-	--execution native
-	--bootnodes "/ip4/127.0.0.1/tcp/30334/p2p/12D3KooWBBUaVWE5SYj3UvnoXojfS8fvPorw5biRDaDQV7XXwCXm"
```
Comment (eduardogr) [inline scripts/localnet.sh:48]: no need for bootnodes anymore?

File: scripts/localnet.sh [hunk 2]
```diff
@@ -6,6 +6,11 @@
 
 FULL_PATH="$SPEC_PATH$CHAIN.json"
 
+if [ ! -d "$SPEC_PATH" ]; then
+  	echo "*** Creating directory ${SPEC_PATH}..."
+	mkdir $SPEC_PATH
```
Comment (eduardogr) [inline scripts/localnet.sh]: looks it has diff number of spaces here

**Comment by Rubberbandits** [comment]
> no need for bootnodes anymore? 

CLI argument `--discover-local` enables a development feature which forces the node to search the local network for peers. You can read about it in the docs [here](https://docs.substrate.io/reference/command-line-tools/node-template/)

**Review by eduardogr — APPROVED** [review]
LGTM!

**Review by shibshib — APPROVED** [review]

**Pull request merged**